crates/proto/src/rr/serial_number.rs

```diff
@@ -0,0 +1,51 @@
+//! Number type to support Serial Number Arithmetics
+
+use std::{cmp::Ordering, ops::Add};
+
+/// Wrapper type to support Serial Number Aritthmetics as defined
+/// in RFC 1982. The signaure fields (expireation, inception) defined in RFC 4034, section 3.1.5
+/// are serial numbers.
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub struct SerialNumber(pub u32);
```

Comment:
I think this commit should be squashed with the next one so that we can see how it is used in the same commit as the one adding the code.

I think this type could be `pub(crate)`, because it is only used in internal code so far?

Reply:
Commits have been squashed. The visibility needs to be `pub` when we use `SerialNumber` in other places, e.g. `sig_expiration()` & `sig_inception()`.

Reply:
Updated the code.